staging/src/k8s.io/apiserver/pkg/registry/rest/create.go

```diff
@@ -90,6 +90,8 @@ func BeforeCreate(strategy RESTCreateStrategy, ctx context.Context, obj runtime.
 	FillObjectMetaSystemFields(objectMeta)
 	if len(objectMeta.GetGenerateName()) > 0 && len(objectMeta.GetName()) == 0 {
 		objectMeta.SetName(strategy.GenerateName(objectMeta.GetGenerateName()))
+	} else {
+		objectMeta.SetGenerateName("")
```

Comment:
the change breaks backward-compatibility. e.g. for now we can create an object w/ non-empty `generateName` and non-empty `name`, when we want to override the name w/ one generated at client-side.

Reply:
@yue9944882 - Thanks for taking a look. Can you please tell me a scenario where we override name with generateName when both of them are non-empty? AFAIk, generateName has no bearing if the name is also present in objectMeta spec, but I may be wrong.